http2.go

```diff
@@ -339,18 +339,7 @@ func validPseudoPath(v string) bool {
 	return (len(v) > 0 && v[0] == '/') || v == "*"
 }
 
-type CloseWithoutResetBuffer interface {
```

Comment:
No need for this anymore?

Reply:
Yes, `CloseWithoutResetBuffer` is used because of standard's Conn will panic when written after it is closed. Now the bug is fixed by https://github.com/cloudwego/hertz/pull/698, therefore I delete the code.